src/pages/[penname]/book/[bookId]/status.tsx

```diff
@@ -0,0 +1,674 @@
+import Image from "next/image";
+import { CategoryPopover } from "@components/action/CategoryPopover";
+import AuthorList from "@components/Book/AuthorList";
+import BookCoverEditable from "@components/Book/BookCoverEditable";
+import { useRouter } from "next/router";
+import { BookOwnerStatus } from "@prisma/client";
+import { api } from "@utils/api";
+import type {
+  GetServerSidePropsContext,
+  InferGetServerSidePropsType,
+} from "next";
+import { getServerAuthSession } from "@server/auth";
+import { createProxySSGHelpers } from "@trpc/react-query/ssg";
+import { createInnerTRPCContext } from "@server/api/trpc";
+import { appRouter } from "@server/api/root";
+import superjson from "superjson";
+import * as z from "zod";
+import { useState } from "react";
+import { zodResolver } from "@hookform/resolvers/zod";
+import useImageUpload from "@hooks/imageUpload";
+import { ToastContainer, toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.min.css";
+import { useForm } from "react-hook-form";
+import { EditButton } from "@components/action/EditButton";
+
+const validationSchema = z.object({
+  title: z
+    .string()
+    .min(1, { message: "Title is required" })
+    .max(100, { message: "Title is too long" }),
+  description: z.string().max(500, { message: "Description is too long" }),
+  author: z.string(),
+});
+
+type ValidationSchema = z.infer<typeof validationSchema>;
+
+export const getServerSideProps = async (
+  context: GetServerSidePropsContext
+) => {
+  const session = await getServerAuthSession(context);
+  const ssg = createProxySSGHelpers({
+    router: appRouter,
+    ctx: createInnerTRPCContext({ session }),
+    transformer: superjson,
+  });
+  const bookId = context.query.bookId as string;
+  await ssg.book.getData.prefetch({ id: bookId });
+
+  return {
+    props: {
+      trpcState: ssg.dehydrate(),
+      session,
+      bookId,
+    },
+  };
+};
+
+type props = InferGetServerSidePropsType<typeof getServerSideProps>;
+
+const StatusPage = ({ bookId }: props) => {
+  const [isEdit, setIsEdit] = useState(false);
+  const {
+    imageData: bookCover,
+    uploadHandler: setBookCover,
+    resetImageData: resetBookCover,
+  } = useImageUpload();
+  const {
+    imageData: bookWallpaper,
+    uploadHandler: setBookWallpaper,
+    resetImageData: resetBookWallpaper,
+  } = useImageUpload();
+  const router = useRouter();
+  const penname = router.query.penname;
+  const utils = api.useContext();
+  const { data: book } = api.book.getData.useQuery({ id: bookId });
+  const { data: categories } = api.category.getAll.useQuery();
+  const [addedCategories, setAddedCategories] = useState(
+    book?.categories.map((data) => data.category) || []
+  );
+  const deleteBook = api.book.delete.useMutation({
+    onSuccess: () => {
+      void utils.book.invalidate();
+    },
+  });
+  const { data: collaborators } = api.user.getBookCollaborators.useQuery({
```

Comment:
prefetch this in getServerSideProps

Reply:
what is this

Reply:
references:
- https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props
- https://trpc.io/docs/nextjs/ssg-helpers